cmd/reset/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! ## `humility reset`
+//!
+//! `humility reset` will hard reset the system using the debug pin
+//!
+//!
+
+use anyhow::{bail, Result};
+use clap::Command as ClapCommand;
+use clap::{CommandFactory, Parser};
+use humility::hubris::*;
+use humility_cmd::{Archive, Args, Command};
+use probe_rs::Probe;
+
+#[derive(Parser, Debug)]
+#[clap(name = "probe", about = env!("CARGO_PKG_DESCRIPTION"))]
+struct ResetArgs {
+    /// Which probe to connect
+    probe_num: Option<usize>,
+}
+
+fn reset(
+    _hubris: &mut HubrisArchive,
+    _args: &Args,
+    subargs: &[String],
+) -> Result<()> {
+    let subargs = ResetArgs::try_parse_from(subargs)?;
+
+    // Get a list of all available debug probes.
+    let probes = Probe::list_all();
+
+    if probes.is_empty() {
+        bail!("No probes found!");
+    }
+
+    let num = subargs.probe_num.unwrap_or(0);
```

Comment:
It feels like it would be more consistent to allow the probe to be specified on the command line (e..g, as we do for `humility probe`), but I think that would necessitate adding a `reset` function to the `Core` trait (one that would only be implemented for `ProbeCore`).  But this would allow one to (e.g.) reset a probe by specifying VID/PID/Serial.

Reply:
I don't think `ProbeCore` is the right place. `ProbeCore` implies an attachment to a chip which I wanted to avoid with `humility reset` for cases where something has truly wedged the chip. I agree about making it consistent with other humility commands so I'll give this some more thought.